packages/flutter/lib/src/services/image_cache.dart

```diff
@@ -53,10 +54,14 @@ class _UrlFetcher implements ImageProvider {
   @override
   Future<ImageInfo> loadImage() async {
     try {
-      return new ImageInfo(
-        image: await decodeImageFromDataPipe(await http.readDataPipe(Uri.base.resolve(_url))),
-        scale: _scale
-      );
+      Uri resolvedUri = Uri.base.resolve(_url);
+      mojo.MojoDataPipeConsumer dataPipe = await http.readDataPipe(resolvedUri);
+      if (dataPipe == null)
+        throw 'Unable to read data from $resolvedUri.';
```

Comment:
to avoid confusion, I recommend `'Unable to read data from: $resolvedUri';` (so the period doesn't get appended to the URL)